azure/scope/machinepool.go

```diff
@@ -330,17 +334,14 @@ func (m *MachinePoolScope) applyAzureMachinePoolMachines(ctx context.Context) er
 }
 
 func (m *MachinePoolScope) createMachine(ctx context.Context, machine azure.VMSSVM) error {
-	if machine.InstanceID == "" {
-		return errors.New("machine.InstanceID must not be empty")
-	}
-
-	if machine.Name == "" {
-		return errors.New("machine.Name must not be empty")
+	parsed, err := azure2.ParseResourceID(machine.ID)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("failed to parse resource id %q", machine.ID))
 	}
 
 	ampm := infrav1exp.AzureMachinePoolMachine{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      m.AzureMachinePool.Name + "-" + machine.InstanceID,
+			Name:      strings.ReplaceAll(parsed.ResourceName, "_", "-"),
```

Comment:
could this break existing machines? just wondering if we're introducing a new way of naming machines that would change exisitng machines